manager/src/main/java/io/gingersnapproject/hotrod/CommandProcessor.java

```diff
@@ -162,10 +164,51 @@ public void putAll(GingersnapHeader header, Map<byte[], byte[]> entryMap) {
             writeSuccess(header);
             return;
          }
+         Map<String, String> transformed = new HashMap<>();
          for (Map.Entry<byte[], byte[]> entry : entryMap.entrySet()) {
-            maps.put(cacheName, toString(entry.getKey()), toString(entry.getValue()));
+            transformed.put(toString(entry.getKey()), toString(entry.getValue()));
          }
-         writeSuccess(header);
+
+         maps.putAll(cacheName, transformed).subscribe()
```

Comment:
Lets do this in a separate PR. This has a bit of noise now.

Reply:
Sure, I'll revert in here and open another.